searchlib/src/tests/features/nns_closeness/nns_closeness_test.cpp

```diff
@@ -0,0 +1,177 @@
+// Copyright Verizon Media. Licensed under the terms of the Apache 2.0 license. See LICENSE in the project root.
+
+#include <vespa/vespalib/testkit/test_kit.h>
+#include <vespa/searchlib/features/setup.h>
+#include <vespa/searchlib/fef/test/indexenvironment.h>
+#include <vespa/searchlib/fef/test/indexenvironmentbuilder.h>
+#include <vespa/searchlib/fef/test/queryenvironment.h>
+#include <vespa/searchlib/features/closenessfeature.h>
+#include <vespa/searchlib/fef/fef.h>
+#include <vespa/searchlib/fef/test/dummy_dependency_handler.h>
+#include <vespa/vespalib/stllike/asciistream.h>
+#include <vespa/vespalib/util/stringfmt.h>
+
+using search::feature_t;
+using namespace search::fef;
+using namespace search::fef::test;
+using namespace search::features;
+using CollectionType = FieldInfo::CollectionType;
+using DataType = FieldInfo::DataType;
+
+const vespalib::string labelFeatureName("closeness(label,nns)");
+const vespalib::string fieldFeatureName("closeness(bar)");
+
+struct BlueprintFactoryFixture {
+    BlueprintFactory factory;
+    BlueprintFactoryFixture() : factory()
+    {
+        setup_search_features(factory);
+    }
+};
+
+struct IndexFixture {
+    IndexEnvironment indexEnv;
+    IndexFixture() : indexEnv()
+    {
+        IndexEnvironmentBuilder builder(indexEnv);
+        builder.addField(FieldType::ATTRIBUTE, CollectionType::SINGLE, DataType::INT64, "foo");
+        builder.addField(FieldType::ATTRIBUTE, CollectionType::SINGLE, DataType::TENSOR, "bar");
+    }
+};
+
+struct FeatureDumpFixture : public IDumpFeatureVisitor {
+    virtual void visitDumpFeature(const vespalib::string &) override {
+        TEST_ERROR("no features should be dumped");
+    }
+    FeatureDumpFixture() : IDumpFeatureVisitor() {}
+};
+
+struct Labels {
```

Comment:
Labels, NoLabels and SingleLabel are now duplicated in 3 unit tests. Please move them to searchlib/src/vespa/searchlib/fef/test and put them in namespace search::fef::test.

Reply:
Labels etc. moved to fef::test now.